### PR TITLE
Enable new permissions system

### PR DESCRIPTION
**Pull request by DanielRyanSmith**
My [old PR](https://github.com/GoogleChrome/chromium-dashboard/pull/2017) for this accidentally got closed after I force-pushed, so I've made this one.

This PR fully enables the permissions system described in https://github.com/GoogleChrome/chromium-dashboard/pull/2011. This PR will enact the new changes, and the application will behave as intended in the original design.

### CODE REVIEW

**Comment by jrobbins** [comment]
Is this ready for review?

File: framework/permissions.py [hunk 1]
```diff
@@ -61,28 +61,43 @@ def can_create_feature(user):
...
 
-def can_edit_feature(user, feature):
-  """Return True if the user is allowed to edit the given feature."""
-  # TODO(jrobbins): make this per-feature
-  if not can_view_feature(user, feature):
+
+def feature_edit_list(user):
+  """Return a list of features the current user can edit"""
+  if not user:
     return False
```
Comment (jrobbins) [inline framework/permissions.py:78]: False is not a list.

File: framework/permissions.py [hunk 2]
```diff
@@ -150,11 +165,20 @@ def check_login(self, *args, **kwargs):
...
 
-def require_edit_feature(handler):
-  """Handler decorator to require the user can edit the current feature."""
-  # TODO(jrobbins): make this per-feature
-  def check_login(self, *args, **kwargs):
-    return _reject_or_proceed(
-        self, handler, args, kwargs, can_edit_any_feature)
+def validate_feature_edit_permission(handler_obj, feature_id):
+  """Check if user has permission to edit feature and abort if not."""
+  """Redirect, abort(403), or call handler_method."""
```
Comment (jrobbins) [inline framework/permissions.py:170]: Double doc strings.

**Review by jrobbins — COMMENTED** [review]
Some quick nits, even if you are not done yet.

File: framework/permissions.py (hunk 1 above)
Comment (DanielRyanSmith) [reply framework/permissions.py]: Thanks @jrobbins. I was having trouble rebasing the changes and dealing with conflicts after all the recent changes, and I might have some issues here. I'm trying to find them all, but perhaps more will be found in manual testing.

**Review by jrobbins — APPROVED** [review]

**Pull request merged**